scripts/maintenance.awk

```diff
@@ -0,0 +1,14 @@
+## Copyright (C) 1996-2019 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+#
+# This script performs easily automated code updates for
+# removal or renaming of deprecated/obsolete syntax.
+# It is run by maintenance script over .h, .c, .cc, and .cci files
+#
+
+{ print }
```

Comment:
I do not understand why we should commit a script that does nothing. Please explain (probably here, in a PR comment, to start with).

Reply:
This is the infrastructure part of scripting away the annoyances like HERE or NULL. Once this is in we can have lots of more targeted discussions about how to replace particular code patterns on an individual basis.

I chose to do this part as a standalone PR so that we can discuss this script change in terms of "how" to do changes rather than get bogged down into another long discussion about "what" of any particular change.